diff --git a/app.js b/app.js
index 8cb3a4f..10f60f1 100644
--- a/app.js
+++ b/app.js
@@ -204,6 +204,7 @@ if (app.get('env') === 'development') {
 
 // App configuration for all environments
 app.set('port', process.env.PORT || 3000);
+app.set('listen_address', config.system.listen_address);
 app.set('views', __dirname + '/views');
 app.set('view engine', 'ejs');
 app.use(favicon(__dirname + '/public/img/favicon.ico'));
@@ -313,8 +314,8 @@ app.use(function (req, res, next) {
 });
 app.use(serveStatic(path.join(__dirname, 'public')));
 
-var server = app.listen(app.get('port'), function () {
-    logger.info('openHAB-cloud: express server listening on port ' + app.get('port'));
+var server = app.listen(app.get('port'), app.get('listen_address'), function () {
+    logger.info('openHAB-cloud: express server listening on port ' + (app.get('listen_address')?app.get('listen_address'):'*') + ':' + app.get('port'));
 });
 
 var io = require('socket.io').listen(server, {
